mozdownload/scraper.py

```diff
@@ -427,12 +427,17 @@ def get_latest_build_date(self):
         # Read status file for the platform, retrieve build id,
         # and convert to a date
         headers = {'Cache-Control': 'max-age=0'}
+
+        retval = None
         r = requests.get(url + parser.entries[-1],
                          auth=self.authentication, headers=headers)
-        r.raise_for_status()
+        try:
+            r.raise_for_status()
+
+            retval = datetime.strptime(r.text.split('\n')[0], '%Y%m%d%H%M%S')
```

Comment:
You can return immediately. There is no need for the extra retval variable. In case of an exception the retval is not used at all.